### DISCUSSION TITLE
Get Images from Websocket : How?

### DISCUSSION THREAD

**Discussion by seghier** [comfyanonymous/ComfyUI#3226] [General]
Hello
Ho we can retrieve the image from **Send Image (WebSocket)** or **SaveImageWebsocket**
I use PyCharm or any other app support Python

![image](https://github.com/comfyanonymous/ComfyUI/assets/6026588/fd4b5078-7698-4c20-955e-157aa7afa04a)

**Comment by comfyanonymous** [comment]
Here's an example: https://github.com/comfyanonymous/ComfyUI/blob/master/script_examples/websockets_api_example_ws_images.py

**Comment by seghier** [comment]
Thank you for the example
I thought that we can use Websocket like Photoshop server to send data and read it in any app 
If it is possible i need a way to send the progress and the preview images 
for example `blob:http://127.0.0.1:8188/47064d6f-6256-422c-8847-124c09738699` 
in realtime from ComfyUI

**Comment by seghier** [comment]
Hello again
I run the workfolw and get the image.
How i can get the image saved or sent by websocket without need to save api format and run it again via Python?

something like this maybe?

```
import websocket #NOTE: websocket-client (https://github.com/websocket-client/websocket-client)
import uuid
import json
import urllib.request
import urllib.parse

server_address = "127.0.0.1:8188"
client_id = str(uuid.uuid4())

def get_images(ws):
    output_images = {}
    current_node = ""

    while True:
        out = ws.recv()
        if current_node == 'save_image_websocket_node':
            images_output = output_images.get(current_node, [])
            images_output.append(out[8:])
            output_images[current_node] = images_output
        
    return output_images

ws = websocket.WebSocket()
ws.connect("ws://{}/ws?clientId={}".format(server_address, client_id))
images = get_images(ws)

print(images)
#Commented out code to display the output images:

for node_id in images:
    for image_data in images[node_id]:
        from PIL import Image
        import io
        image = Image.open(io.BytesIO(image_data))

        image.show()
```

**Reply by ishaangupta2311** [reply]
did you figure it out ? i am also facing issue with retrieving image from the comfyui server
